lib/launchers/ocm.rb

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env ruby
+
+lib_path = File.expand_path(File.dirname(File.dirname(__FILE__)))
+unless $LOAD_PATH.any? {|p| File.expand_path(p) == lib_path}
+  $LOAD_PATH.unshift(lib_path)
+end
+
+require 'common'
+require 'json'
+require 'yaml'
+require 'tmpdir'
+require 'git'
+
+module BushSlicer
+  class OCM
+    include Common::Helper
+
+    attr_reader :config
+    attr_reader :token, :token_file, :url, :region, :version, :nodes, :lifespan, :cloud, :cloud_opts, :multi_az, :aws_account_id, :aws_access_key, :aws_secret_key
+
+    def initialize(**options) 
+      service_name = ENV['OCM_SERVICE_NAME'] || options[:service_name] || 'ocm'
+      @opts = default_opts(service_name)&.merge options
+      unless @opts
+        @opts = options
+      end
+
+      # OCM token is mandatory
+      # it can be defined by token or by token_file
+      @token = ENV['OCM_TOKEN'] || @opts[:token]
+      @token_file = @opts[:token_file]
+      unless @token
+        if @token_file
+          token_file_path = expand_private_path(@token_file)
+          @token = File.read(token_file_path)
+        else
+          raise "You need to specify OCM token by 'token' or by 'token_file'"
+        end
+      end
+
+      # region is mandatory
+      # in the future we can extend support for other clouds, e.g. GCP and ARO
+      @region = ENV['OCM_REGION'] || ENV['AWS_REGION'] || @opts[:region]
+
+      # url defines the OCM environment (prod, integration or stage)
+      # currently, the url is ignored as many teams use the stage environment
+      @url = ENV['OCM_URL'] || @opts[:url] || 'https://api.stage.openshift.com'
+
+      # openshift version is optional
+      @version = ENV['OCM_VERSION'] || ENV['OCP_VERSION'] || @opts[:version]
+
+      # number of worker nodes
+      # minimum is 2
+      # default value is 4
+      @nodes = ENV['OCM_NODES'] || @opts[:nodes]
+
+      # lifespan in hours
+      # default value is 24 hours
+      @lifespan = ENV['OCM_LIFESPAN'] || @opts[:lifespan]
+
+      # multi_az is optional
+      # default value is false
+      @multi_az = ENV['OCM_MULTI_AZ'] || @opts[:multi_az]
+
+      # BYOC (Bring Your Own Cloud)
+      # you can refer to already defined cloud in config.yaml
+      # currently, only AWS is supported
+      if ENV['AWS_ACCOUNT_ID'] && ENV['AWS_ACCESS_KEY'] && (ENV['AWS_SECRET_ACCESS_KEY'] || ENV['AWS_SECRET_KEY'])
+        @aws_account_id = ENV['AWS_ACCOUNT_ID']
```

Comment:
Are you sure OCM requires account_id? This is the first tool I see to request that. I commented about it above but still.

Reply:
Yes, it is. This is the error when I omit it
```
Attribute 'aws.account_id' is mandatory on BYOC clusters
```